packages/mui-material/src/StepButton/StepButton.test.js

```diff
@@ -145,4 +146,66 @@ describe('<StepButton />', () => {
 
     expect(getByRole('button')).not.to.equal(null);
   });
+
+  describe('prop: StepIconComponent', () => {
+    it('should render', () => {
+      function CustomizedIcon() {
+        return <div data-testid="custom-icon" />;
+      }
+      const { container, getByTestId } = render(
+        <Step active completed>
+          <StepButton StepIconComponent={CustomizedIcon}>Step One</StepButton>
+        </Step>,
+      );
+
+      const icon = container.querySelector(`.${stepLabelClasses.iconContainer}`);
+
+      getByTestId('custom-icon');
+      expect(icon).not.to.equal(null);
+      expect(icon).not.to.have.attribute('data-testid').equal('CheckCircleIcon');
+    });
+
+    it('should not render', () => {
+      const { container } = render(
+        <Step active completed>
+          <StepButton>Step One</StepButton>
+        </Step>,
+      );
+
+      const icon = container.querySelector(`.${iconClasses.root}`);
+      expect(icon).to.equal(null);
+    });
+
+    describe('renders <StepIcon> with the className completed', () => {
+      it('renders with completed className when completed', () => {
+        function CustomizedIcon() {
+          return <div data-testid="custom-icon" />;
+        }
+        const { container } = render(
+          <Step completed>
+            <StepButton StepIconComponent={CustomizedIcon}>Step One</StepButton>
+          </Step>,
+        );
+  
+        const icon = container.querySelector(`.${stepLabelClasses.iconContainer}`);
+        expect(icon).to.have.class(stepLabelClasses.completed);
+      });
+    });
+
+    describe('renders <StepIcon> with the className active', () => {
+      it('renders with active className when active', () => {
+        function CustomizedIcon() {
+          return <div data-testid="custom-icon" />;
+        }
+        const { container } = render(
+          <Step active>
+            <StepLabel StepIconComponent={CustomizedIcon}>Step One</StepLabel>
+          </Step>,
+        );
+  
+        const icon = container.querySelector(`.${stepLabelClasses.iconContainer}`);
+        expect(icon).to.have.class(stepLabelClasses.active);
+      });
+    });
```

Comment:
These tests are unnecessary. Instead add tests for `StepIconProps` through which the passed prop is applied to the icon.

Reply:
Apologies for the delayed response. @ZeeshanTamboli

The `StepIconProps` is a set of properties that can be passed to the `StepButton` component to customize the appearance and behavior of the step icon. Following the `StepLabel` implementation and test. The `StepIconComponent` is the prop that can be used in the Stepper component to customize the appearance of step icons in the step navigation. It allows us to replace the default step icons with our own custom icons or components.

Reply:
@gitstart, I'm aware of that. What I'm suggesting is that the tests from line 179 to 209 are redundant. The previous tests using `StepIconComponent` already cover this. I recommend adding tests for `StepIconProps` instead.

> The StepIconProps is a set of properties that can be passed to the StepButton component to customize the appearance and behavior of the step icon.

Yes. This is what I meant to test.